eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -192,8 +195,11 @@ stages:
                   Location: ${{ coalesce(parameters.Location, cloud.value.Location) }}
                   Cloud: ${{ cloud.key }}
 
-  # The Prerelease and Release stages are conditioned on whether we are building a pull request and the branch.
-  - ${{if and(in(variables['Build.Reason'], 'Manual', ''), not(contains(variables['Build.DefinitionName'], 'weekly')), eq(variables['System.TeamProject'], 'internal'))}}:
+  # The Prerelease and Release stages are conditioned on:
+  # 1. Internal trigger
+  # 2. Not weekly build
+  # 3. Manual trigger (include manual defined Build.Reason param with empty) / Non-PR trigger with 'AutoRelease' flag
+  - ${{if and(or(in(variables['Build.Reason'], 'Manual', ''), and(ne(variables['Build.Reason'], 'PullRequest'), eq(parameters.AutoRelease, 'true'))), not(contains(variables['Build.DefinitionName'], 'weekly')), eq(variables['System.TeamProject'], 'internal'))}}:
```

Comment:
This condition is definitely starting to become impossible to understand. @benbp can you give more context on why you made the change in https://github.com/Azure/azure-sdk-for-go/pull/17312? 

Should we consider doing auto release check behavior for all go libraries both mgmt and data-plane?

Reply:
@weshaggard I made this change based on an ask from @RickWinter and @seankane-msft to hide the release stage for builds that should not release, with the idea that it would make release tracking/pipeline history easier to read, reduce confusion, and be consistent with the pipeline UI for other languages.

I think adding `AutoRelease` does enable it for all go libraries, where the data plane ones just default to false and so don't need to specify it. We would still want to do the other checks here like build reason, etc. if we wanted to preserve the release stage hiding behavior.

Reply:
I do think it might make it a little easier to understand and read if we can put some of the condition as the default value for. Perhaps we just keep the internal and AutoRelease condition here and move the others to the default value.

Also I think AutoRelease is a little misleading, we should probably name this to IncludeRelease.

Reply:
I don't quite get the idea of the default value. Is that mean we just use another variable to judge other condition except internal and AutoRelease?

Reply:
I agree re `IncludeRelease` naming because having the release included doesn't guarantee that it will be auto released due to the "verify release" stage.

We were chatting offline and @weshaggard had the idea that maybe we could improve the readability of the conditionals by using nested ifs (since I don't think devops supports multi-line expressions). The downside is that the template below has a higher indentation level, but I think that's fine.

```
  - ${{ if and(ne(variables['Build.Reason'], 'PullRequest'], eq(variables['System.TeamProject'], 'internal')) }}:
    - ${{ if not(contains(variables['Build.DefinitionName'], 'weekly')) }}:
      - ${{ if or(eq(variables['Build.Reason'], 'Manual'), eq(${{ parameters.IncludeRelease }}, true)) }}:
```

By "default value" we were thinking about trying to declare some of the conditional expressions in a parameter declaration as the default value, but I'm not sure devops would evaluate that properly at template compile time.

Reply:
Looks great. I've changed the PR according to this. BTW, the final if for `Manual` part, I use `in(variables['Build.Reason'], 'Manual', '')` to keep the same as your previous logic.

Reply:
One single test: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1460469&view=results